pkg/kmm/kmm.go

```diff
@@ -18,207 +18,267 @@ import (
 
 const assetKey string = "kmm-asset-key"
 const assetLockKey string = "kmm-asset-lock"
+const defaultBackOff time.Duration = 20 * time.Second
 
-// Config - the complete configuration provided for all kmm use
-type Config struct {
-	KubeadmCfg kubeadm.Config
-
-	/*
-	 To provide the least change to kubeadm and prevent access to the key,
-	 we copy and link from the persistent source as appropriate
+// Interface defined to enable testing of core functions without dependencies
+type Interface interface {
+	CleanUp(releaseLock, deleteAssets bool) (err error)
+	CopyKubeCa() (err error)
+	InstallNetwork() (err error)
+	TokensDeploy() error
+	UpdateCloudCfg() (err error)
+	CreateAndStartKubelet(master bool) error
+}
 
-	 TODO: Update kubeadm to allow for specified CA key locations
-	*/
+// ConfigType is the complete configuration provided for all kmm use
+type ConfigType struct {
+	KubeadmCfg           *kubeadm.Config
 	KubePersistentCaCert string
 	KubePersistentCaKey  string
 	ClusterName          string
 	NetworkProvider      string
+	MasterBackOffTime    time.Duration
+	ExitOnCompletion     bool
+	Etcd                 etcd.Clienter
+	Kubeadm              kubeadm.Kubeadmer
+	Kmm                  Interface
 }
 
-// Manifests - Will generate static manefest files
-func Manifests(cfg kubeadm.Config) (err error) {
-	if err = kubeadm.WriteManifests(cfg); err != nil {
+// Both structs here use the same config but are bound to different methods...
+
+// Config is tied to the Primary methods (no interface - not for mocking)
+type Config struct {
```

Comment:
Maybe a better name required but essentially there are high level methods (entry-points) and then low-level methods (mockable bits) - all of which use the same config. This allows a simple type cast to move the data...